sql/src/main/java/io/crate/cluster/gracefulstop/DecommissioningService.java

```diff
@@ -199,6 +202,14 @@ protected void doStart() {
 
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
+        if (event.metaDataChanged()) {
+            if (GRACEFUL_STOP_REALLOCATE_SETTING.setting().get(event.previousState().metaData().settings()).booleanValue() !=
```

Comment:
Is this still needed? Wouldn't the setting framework log a deprecation warning anyhow on a change?

Reply:
hmm, seems that it logs only the first time the setting is accessed.

When I tested that, I had changed the initial value on the configuration file before starting the node, so it was indeed reported at node startup. Then I tested changing the setting with the `SET` statement (since it's a runtime setting) and no log appeared which l confused me into that it's not picked up at state change - therefore I left that code thinking that this is required.

So If I remove this, it's not reported on every change of the setting but only the first time is it changed/accessed. However, I can see that we handle all deprecated settings this way, thus it makes sense to align. I'll remove it.

Thanks for the comment